src/Node.cpp

```diff
@@ -66,4 +66,17 @@ Node::RemoveFromParents(GroupWeak& aParent, Node& aChild) {
   }
 }
 
+bool
+Node::Traverse(const NodePtr& aRootNode, const TraverseFunction& aTraverseFunction) {
+  if (aTraverseFunction(aRootNode, {})) {
```

Comment:
I think I would prefer we use nullptr instead of {} if for no other reason than consistency with the rest of the code (I personally find it easier to read but that is mostly just preference).

Reply:
Ok, PR updated to use nullptr